include/dlaf/communication/kernels.h

```diff
@@ -68,16 +68,53 @@ matrix::Tile<const T, D> recvBcastAlloc(TileElementSize tile_size, comm::IndexT_
 template <class T, Device D, template <class> class Future>
 void scheduleSendBcast(const comm::Executor& ex, Future<matrix::Tile<const T, D>> tile,
                        hpx::future<common::PromiseGuard<comm::Communicator>> pcomm) {
-  hpx::dataflow(ex, hpx::util::unwrapping(sendBcast_o), internal::prepareSendTile(std::move(tile)),
+  hpx::dataflow(ex, matrix::unwrapExtendTiles(sendBcast_o), internal::prepareSendTile(std::move(tile)),
                 std::move(pcomm));
 }
 
+// This implements scheduleRecvBcast for when the device of the given tile and
+// the communication device are different. In that case we make use of
+// recvBcastAlloc, which will allocate a tile on the correct device for
+// communication, receive the data, and return the tile. When communication is
+// ready, we copy the data from the tile on the communication device to the tile
+// passed into scheduleRecvBcast.
+template <Device D, Device CommunicationD>
+struct scheduleRecvBcastImpl {
+  template <class T>
+  static void call(const comm::Executor& ex, hpx::future<matrix::Tile<T, D>> tile,
```

Comment:
@rasolca @msimberg

This is something that I'd like to know, since I'm working on a PR with similar functions.

Do we want to return a `Future` to the tile with the result of this function for enabling the user to (possibly) attach a continuation?

Reply:
I don't really have an answer to this, except that I'd leave it void until it's needed. This is again about the conflict between the Matrix handling lifetimes and access to tiles vs tiles being passed as arguments and being returned from tasks. When one is using only one of the methods things are mostly clean but when they get mixed things get messy.

Reply:
Sorry, it wasn't very clear what I meant. I should had pointed out these lines

https://github.com/eth-cscs/DLA-Future/blob/505e99981434189a06441f232ea3aa8d761920d6/include/dlaf/communication/kernels.h#L91-L93

where a temporary tile is allocated and used.

It is true that you are using it for the communication and then it is eventually copied back to the user passed one. I was just wondering if we can rely on the allocator (e.g. umpire) for these kind of temporary tiles or not.